File: examples/ruby/tags/id/tag_id_search.rb
# -*- coding: utf-8 -*-
# 
# Esimerkki ruby koodin pätkä, jolla haetaan ja tuloste-
# taan id:llä haetun tagin kysymykset.
#
# Tekijä: Jarkko Moilanen (@kyyberi)
# Päivämäärä: 13.7.2013
# Lisenssi: http://creativecommons.org/licenses/by-sa/3.0/
# 
# -------------------------------------------------------

# Tarvittavat gemit, asenna tarvittaessa tyyliin: 
# "gem install json"
require 'open-uri'
require 'json'


# 1. Ensin haetaan ID tagien listauksesta

# URL mistä tag lista haetaan
@murl = 'http://api.avoindata.net/tags'
# hae JSON sisältö
@json_contents  = open(@murl) {|f| f.read }
# parsi JSON
@content = JSON.parse(@json_contents)
# ota toisen objektin id talteen
@needed_id = @content['tags'][1]['wordid']

# 2. haetaan kyseisen id:n omaavan tagin kysymykset

# URL mistä tag lista haetaan
@murl = 'http://api.avoindata.net/tags/id/' + @needed_id.to_s 
# hae JSON sisältö
@json_contents  = open(@murl) {|f| f.read }
# parsi JSON
@content = JSON.parse(@json_contents)

# iteroi tietojoukko
@content[@needed_id.to_s].each do |item|
  ititle = item['title']
  # tulosta tai mitä sitten haluatkaaan tehdä
  puts ititle
end

File: examples/php/tags/id/tag_id_search.php
<?php
/* #########################################
   PHP esimerkki miten tagien nimet ja niihin
   liittyvien kysymysten määrät luetaan

   Jarkko Moilanen
   11.7.2013
   Lisenssi: http://creativecommons.org/licenses/by-sa/3.0/

*/

$search_id = "25";
$json_url = "http://api.avoindata.net/tags/id/".$search_id;
$json = file_get_contents($json_url);

$obj = json_decode($json);
$tags = $obj->{$search_id};

foreach ($tags as $tag) {
  echo $tag->title;
  echo $tag->updated;
}

?>
